rpcclient/infrastructure.go

```diff
@@ -1524,3 +1557,65 @@ func (c *Client) BackendVersion() (BackendVersion, error) {
 
 	return *c.backendVersion, nil
 }
+
+// Batch makes batch requests
+func (c *Client) Batch() *Client {
+	c.batch = true //copy the client with changed batch setting
+	c.start()
+	return c
+}
+
+func (c *Client) sendAsync() FutureGetBulkResult {
+	// convert the array of marshalled json requests to a single request we can send
+	responseChan := make(chan *response, 1)
+	marshalledRequest := []byte("[")
+	for iter := c.batchList.Front(); iter != nil; iter = iter.Next() {
+		request := iter.Value.(*jsonRequest)
+		marshalledRequest = append(marshalledRequest, request.marshalledJSON...)
+		marshalledRequest = append(marshalledRequest, []byte(",")...)
+	}
+	if len(marshalledRequest) > 0 {
+		marshalledRequest = marshalledRequest[:len(marshalledRequest)-1]
+	}
```

Comment:
Deserves a comment explaining that this piece of code removes the last hanging `,` character.

Reply:
fixed in 9c315730